analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -59,28 +58,76 @@ protected RequestsWithExcessiveLengthBase(System.Resources.ResourceManager rspec
         }
 
         protected override void Initialize(ParameterLoadingAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer,
+            context.RegisterCompilationStartAction(
                 c =>
                 {
-                    if (!IsEnabled(c.Options))
-                    {
-                        return;
-                    }
-
-                    if (c.Node is TAttributeSyntax attribute
-                        && (attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_DisableRequestSizeLimitAttribute, c.SemanticModel)
-                            || IsInvalidRequestSizeLimit(attribute, c.SemanticModel)
-                            || IsInvalidRequestFormLimits(attribute, c.SemanticModel)))
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, c.Node.GetLocation()));
-                    }
-                },
-                Language.SyntaxKind.Attribute);
+                    var attributesOverTheLimit = new ConcurrentDictionary<SyntaxNode, Attributes>();
+
+                    c.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer,
+                        cc =>
+                        {
+                            if (!IsEnabled(c.Options))
+                            {
+                                return;
+                            }
+
+                            var attribute = (TAttributeSyntax)cc.Node;
+
+                            if (attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_DisableRequestSizeLimitAttribute, cc.SemanticModel))
+                            {
+                                cc.ReportDiagnosticWhenActive(Diagnostic.Create(rule, attribute.GetLocation()));
+                            }
+
+                            if ((IsInvalidRequestSizeLimit(attribute, cc.SemanticModel)
+                                 || IsInvalidRequestFormLimits(attribute, cc.SemanticModel))
+                                && GetMethodOrClassDeclaration(attribute, cc.SemanticModel) is { } delcaration)
+                            {
+                                var isRequestFormLimits = attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_RequestFormLimitsAttribute, cc.SemanticModel);
```

Comment:
you are doing this query twice (once in the `if`, then here) and this is quite an expensive call (whenever you use the SemanticModel, think : "expensive performance cost").

Could you, when doing the `Is<Attribute>` call , extrac the `TAttributeSyntax` node , and then based on what's null or not null , create the `Attribute`?